### PR TITLE
Updated nav menu for docs

### PR DESCRIPTION
**Pull request by Philinphiladelphia**
## Issue Number

N/A

## Purpose/Implementation Notes

This PR adds API docs for the following endpoints: 
- Attachments
- Material Requests
- Organization User Settings

It also updates the nav menu to reflect all of these files.

Additional changes made just as I saw them:
- Made several more fields read-only on `material-request`.
- Automatically set the `requester` field on material-request to the user currently signed in.
- Fixed a typo or two.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- New Feature

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/api/attachments.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+```json
+Content-Type application/json
+201 CREATED
+
+{
+   "id":18,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
```
Comment (kurtwheeler) [inline docs/api/attachments.md:39]:
```suggestion
   "s3_bucket":"bucket-name",
   "s3_key":"keyname",
```

File: docs/api/attachments.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+```json
+Content-Type application/json
+200 OK
+
+{
+   "id":21,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
```
Comment (kurtwheeler) [inline docs/api/attachments.md:72]:
```suggestion
   "s3_bucket":"bucket-name",
   "s3_key":"keyname",
```

File: docs/api/attachments.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+Content-Type application/json
+200 OK
+
+...
+   {
+      "id":21,
+      "filename":"attachment_file",
+      "description":"A file for testing",
+      "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+      "s3_key":"s3 key",
```
Comment (kurtwheeler) [inline docs/api/attachments.md:103]:
```suggestion
   "s3_bucket":"bucket-name",
   "s3_key":"keyname",
```

File: docs/api/attachments.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+`PUT/PATCH` `/attachments/:id`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The url of the s3_bucket where your attachment file is stored.
```
Comment (kurtwheeler) [inline docs/api/attachments.md:125]: This should just be the bucket name of the attachment file. The full URL can be constructed with the bucket and the key, but the S3 client wants them as parameters.

File: docs/api/attachments.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+```json
+Content-Type application/json
+200 OK
+
+{
+   "id":21,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
```
Comment (kurtwheeler) [inline docs/api/attachments.md:149]:
```suggestion
   "s3_bucket":"bucket-name",
   "s3_key":"keyname",
```

File: docs/api/attachments.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+`POST` `/attachments/`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The url of the s3_bucket where your attachment file is stored.
```
Comment (kurtwheeler) [inline docs/api/attachments.md:19]: This should also be changed to say bucket name.

File: docs/api/attachments.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+Content-Type application/json
+201 CREATED
+
+{
+   "id":18,
+   "filename":"attachment_file",
+   "description":"A file for testing",
+   "s3_bucket":"https://bucket-name.s3.region.amazonaws.com/keyname",
+   "s3_key":"s3 key",
+   "deleted":"None",
```
Comment (kurtwheeler) [inline docs/api/attachments.md:40]: Is this the safedelete field or `s3_resource_deleted`? I think it maybe is the former but is intended to be the latter?

File: docs/api/material_requests.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+Name                           | Type   | Description
+-------------------------------|--------|---
+material                       | int    | The ID of the material being requested.
+requester_signed_mta_attachment| int    | The ID of an attachment object which references a signed Material Transfer Agreement attachment.
+irb_attachment                 | int    | The ID of an attachment object which references an Institutional Review Board attachment.
+
+
+*Note:*
+
+- `sequence_map_for` is optional, and should only be provided when the attachment is a sequence map.
```
Comment (kurtwheeler) [inline docs/api/material_requests.md:24]: This is out of place.

File: docs/api/material_requests.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+Content-Type application/json
+201 CREATED
+
+{
+   "id":6,
+   "material":5,
+   "requester":"ee1d43f8-7407-4031-9f1b-f9a5fcf2da26",
+   "requester_signed_mta_attachment":15,
+   "irb_attachment":14,
+   "executed_mta_attachment":13,
```
Comment (kurtwheeler) [inline docs/api/material_requests.md:39]: I wouldn't expected `executed_mta_attachement` to be populated after the above POST request.

File: docs/api/material_requests.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+    "id":6,
+    "material":5,
+    "requester":"ee1d43f8-7407-4031-9f1b-f9a5fcf2da26",
+    "requester_signed_mta_attachment":15,
+    "irb_attachment":14,
+    "executed_mta_attachment":13,
+    "is_active":true,
+    "status":"PENDING",
+    "assigned_to":"f75bd715-e188-4c4e-80a3-623793770730"
+    }
```
Comment (kurtwheeler) [inline docs/api/material_requests.md:106]: Shouldn't this be wrapped in square brackets to denote that it's a list.

File: docs/api/material_requests.md (hunk 9 above)
Comment (kurtwheeler) [reply docs/api/material_requests.md:39]: Err, well I mean after someone creates it, at least the executed MTA will be null.

File: docs/api/user_settings.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+   "request_assigned_notif":true,
+   "reminder_notif":true,
+   "user":"UUID("   "8adbfcf0-7b4e-45b7-811a-f63a44d63c3d"   ")",
+   "organization":3,
+   "created_at":"2020-07-01T19:52:07+0000",
+   "updated_at":"2020-07-01T19:52:07+0000"
+}
+```
+
+## Update an attachment
```
Comment (kurtwheeler) [inline docs/api/user_settings.md:38]: It's not an attachment.

File: docs/api/attachments.md (hunk 7 above)
Comment (Philinphiladelphia) [reply docs/api/attachments.md:40]: Good catch. I just changed the serializer.

File: docs/api/material_requests.md (hunk 8 above)
Comment (Philinphiladelphia) [reply docs/api/material_requests.md:24]: Good catch.

File: docs/api/material_requests.md (hunk 9 above)
Comment (Philinphiladelphia) [reply docs/api/material_requests.md:39]: That is true. I'll set them to null in the POST response.

File: docs/api/material_requests.md (hunk 10 above)
Comment (Philinphiladelphia) [reply docs/api/material_requests.md:106]: Done.

File: docs/api/user_settings.md (hunk 11 above)
Comment (Philinphiladelphia) [reply docs/api/user_settings.md:38]: Changed.

File: docs/api/attachments.md (hunk 6 above)
Comment (Philinphiladelphia) [reply docs/api/attachments.md]: Done.

File: docs/api/attachments.md (hunk 1 above)
Comment (Philinphiladelphia) [reply docs/api/attachments.md]: Done.

File: docs/api/attachments.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+
+`POST` `/attachments/`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The bucket name if the attachment file.
```
Comment (kurtwheeler) [inline docs/api/attachments.md:19]:
```suggestion
s3_bucket       | string | The bucket name of the attachment file.
```

File: docs/api/attachments.md [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+
+`PUT/PATCH` `/attachments/:id`
+
+Parameters:
+
+Name            | Type   | Description
+----------------|--------|---
+filename        | string | The filename of the attachment.
+description     | string | A description of the attachment.
+s3_bucket       | string | The bucket name if the attachment file.
```
Comment (kurtwheeler) [inline docs/api/attachments.md:125]:
```suggestion
s3_bucket       | string | The bucket name of the attachment file.
```

**Pull request merged**